src/theory/datatypes/theory_datatypes_utils.h

```diff
@@ -245,6 +245,46 @@ Node sygusToBuiltin(Node c, bool isExternal = false);
  */
 Node sygusToBuiltinEval(Node n, const std::vector<Node>& args);
 
+/** Get free symbols in a sygus datatype type
+ *
+ * Add the free symbols (expr::getSymbols) in terms that can be generated by
+ * sygus datatype sdt to the set syms. For example, given sdt encodes the
+ * grammar:
+ *   G -> a | +( b, G ) | c | e
+ * We have that { a, b, c, e } are added to syms. Notice that expr::getSymbols
+ * excludes variables whose kind is BOUND_VARIABLE.
+ */
+void getFreeSymbolsSygusType(TypeNode sdt,
```

Comment:
Can you use it in the abduction module?

Reply:
I think I could potentially use this as an optimization.  This will require further changes though. I'd prefer doing this in a separate PR.